packages/flutter/lib/src/material/paginated_data_table.dart

```diff
@@ -471,6 +471,9 @@ class PaginatedDataTableState extends State<PaginatedDataTable> {
                     sortColumnIndex: widget.sortColumnIndex,
                     sortAscending: widget.sortAscending,
                     onSelectAll: widget.onSelectAll,
+                    // Make sure no decoration is set on the DataTable
+                    // from the theme, as its already wrapped in a Card.
+                    decoration: const BoxDecoration(),
```

Comment:
This is a bit tricky, as we've discussed offline. It does seem sensible, since the decoration inherited by the DataTable within the Card is unlikely to blend with the Card's own shape and border. The API doc for PaginatedDataTable needs to make this detail clear.

Reply:
I added some docs now to the PaginatedDataTable to explain this.